tests/Unit/Core/Domain/ValueObject/MoneyTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Unit\Core\Domain\ValueObject;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\Decimal\Number;
+use PrestaShop\PrestaShop\Core\Domain\Currency\ValueObject\CurrencyId;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\Money;
+
+class MoneyTest extends TestCase
+{
+    public function testCreateWithValidValue()
+    {
+        $money = new Money(new Number('100'), new CurrencyId(10));
+
+        $this->assertEquals(100, $money->getAmount());
```

Comment:
getAmount returns a Number, not a Integer ;)
```suggestion
        $this->assertEquals('100', (string) $money->getAmount());
```

Reply:
I knew it 😆 I thought to myself "do I need to run such a simple test before the PR ? naaaaah I'm fine"

I got punished by fate 😉